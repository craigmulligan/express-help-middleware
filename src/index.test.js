const app = require('../example')
const request = require('supertest')

describe('middleware', () => {
  test('returns help message with ?help', () => {
    return request(app)
      .get('/greeting/craig?help')
      .expect(200)
      .then(response => {
        expect(response.text).toMatchSnapshot()
      })
  })

  test('returns help message with ?h', () => {
    return request(app)
      .get('/greeting/craig?help')
      .expect(200)
      .then(response => {
        expect(response.text).toMatchSnapshot()
      })
  })

  test('returns normal response without ?help', () => {
    return request(app)
      .get('/greeting/craig')
      .expect(200)
      .then(response => {
        expect(response.text).toBe('Hello craig!')
      })
  })

  test('if it does not match a route it should return all subroutes', () => {
    return request(app)
      .get('/gretting/craig?help')
      .expect(404)
      .then(response => {
        expect(response.text).toMatchSnapshot()
      })
  })

  test('works with post', () => {
    return request(app)
      .post('/greeting?help')
      .expect(200)
      .then(response => {
        expect(response.text).toMatchSnapshot()
      })
  })
})
