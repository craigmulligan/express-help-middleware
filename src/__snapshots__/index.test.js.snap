// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`middleware if it does not match a route it should return all subroutes 1`] = `
"Welcome to my greeting API
	

[GET]:/greeting/:name - Returns a greeting
	

[POST]:/greeting - Creates a greeting
    
      body: 
        name: <string>
        langauge: <string> default english
	
"
`;

exports[`middleware returns help message with ?h 1`] = `
"Returns a greeting
	    params:
      name: A word that would would like to be called by
"
`;

exports[`middleware returns help message with ?help 1`] = `
"Returns a greeting
	    params:
      name: A word that would would like to be called by
"
`;

exports[`middleware works with post 1`] = `
"Creates a greeting
    
      body: 
        name: <string>
        langauge: <string> default english
	
"
`;
